include/h2o/socket/evloop.h

```diff
@@ -34,8 +34,10 @@
 #define H2O_SOCKET_FLAG_IS_CONNECTING 0x40
 #define H2O_SOCKET_FLAG_IS_ACCEPTED_CONNECTION 0x80
 #define H2O_SOCKET_FLAG_IS_CONNECTING_CONNECTED 0x100
+#define H2O_SOCKET_FLAG_DONT_NONBLOCK 0x200
+#define H2O_SOCKET_FLAG_IS_CLOSED 0x400
```

Comment:
I introduced the `H2O_SOCKET_FLAG_IS_CLOSED` state in order to keep the socket around in the case where we need to retry disposal (ssl destroy may send a neverbleed transaction to destroy a cert)

Reply:
Oh I see.

IIUC, you are saying that functions like `SSL_free` decrements the reference count of an RSA key, and that in turn might invoke `priv_rsa_finish` of neverbleed, causing an IPC exchange.

That behavior is annoying.

As we move to pipelining, I wonder if we could get rid of this wait forever, regardless of that wait being blocking or non-blocking. For example, we could get rid of the response for the "del_rsa_key" request. If we change as such, we'd no longer need to wait for the response. And in case the message could not be sent immediately, following messages can push the "del_rsa_key" message in queue onto the connection.

Reply:
Yes, that's correct, it is annoying...

Yes I believe this to be true for pipelining. And that may be a sufficient solution if errors should not happen in the deletion path (which _may_ be true as well, only errors are: via parsing, invalid index and if the key isn't available)